src/coreclr/jit/objectalloc.cpp

```diff
@@ -1334,3 +1482,1724 @@ void ObjectAllocator::RewriteUses()
+                        JITDUMP("Looks like enumerator var re-use (multiple defining GDVs)\n");
+                    }
+
+                    // We will query this info if we see CALL(enumeratorLocal)
+                    // during subsequent analysis, to verify that access is
+                    // under the same guard conditions.
+                    //
+                    CompAllocator alloc(comp->getAllocator(CMK_ObjectAllocator));
+                    CloneInfo*    info    = new (alloc) CloneInfo();
+                    info->m_local         = enumeratorLocal;
+                    info->m_type          = clsHnd;
+                    info->m_pseudoLocal   = pseudoLocal;
+                    info->m_appearanceMap = new (alloc) EnumeratorVarMap(alloc);
+                    info->m_allocBlock    = block;
+                    info->m_allocStmt     = stmt;
+                    info->m_allocTree     = data;
+                    info->m_domBlock      = controllingGDV.m_block;
+                    m_CloneMap.Set(pseudoLocal, info);
+
+                    JITDUMP("Enumerator allocation [%06u]: will track accesses to V%02u guarded by type %s via P%02u\n",
+                            comp->dspTreeID(data), enumeratorLocal, comp->eeGetClassName(clsHnd), pseudoLocal);
+
+                    // If this is not a direct assignment to the enumerator var we also need to
+                    // track the temps that will appear in between. Later we will rewrite these
+                    // to a fresh set of temps.
+                    //
+                    if (lclNum != enumeratorLocal)
+                    {
+                        CheckForEnumeratorUse(enumeratorLocal, lclNum);
+                        RecordAppearance(lclNum, block, stmt, use);
+                    }
+                }
+                else
+                {
+                    JITDUMP(
+                        "Enumerator allocation [%06u]: enumerator type %s cannot be stack allocated, so not tracking enumerator local V%02u\n",
+                        comp->dspTreeID(data), comp->eeGetClassName(clsHnd), enumeratorLocal);
+                }
+            }
+            else
+            {
+                // This allocation is not currently of interest
+                //
+                JITDUMP("Allocation [%06u] was not flagged for conditional escape tracking\n", comp->dspTreeID(data));
+            }
+        }
+        else
+        {
+            // This allocation was not done under a GDV guard
+            //
+            JITDUMP("Allocation [%06u] is not under a GDV guard\n", comp->dspTreeID(data));
+        }
+    }
+    else if (data->OperIs(GT_LCL_VAR, GT_BOX))
+    {
+        // See if we are copying from one enumerator-referring local to another.
+        // This need not be under any guard.
+        //
+        unsigned srcLclNum = BAD_VAR_NUM;
+        if (data->OperIs(GT_BOX))
+        {
+            srcLclNum = data->AsBox()->BoxOp()->AsLclVarCommon()->GetLclNum();
```

Comment:
can you remind me, the GT_BOX can't be on top of anything but LCL_VAR ?

Reply:
Right, the operand must be a LCL_VAR